### DISCUSSION TITLE
Mariadb uuid field in 10.7

### DISCUSSION THREAD

**Discussion by chrisemke** [sqlalchemy/sqlalchemy#7182] [Usage Questions]
I would like to know how I can use the uuid field type of mariadb without having to use a string field.
Maybe analyze the possibility of adding support to the mariadb database using the [official connector](https://pypi.org/project/mariadb/)

**Comment by CaselIT** [comment]
Hi,

Currently mariadb does not support it. From here https://mariadb.com/kb/en/uuid-data-type/ it seems the next version 10.7 will, so in time it will get added to sqlalchemy.

**Comment by zzzeek** [comment]
if you are using a pre-release of 10.7 you can make a UUID type using [UserDefinedType](https://docs.sqlalchemy.org/en/14/core/custom_types.html#creating-new-types)

```py

class UUID(UserDefinedType):

    def get_col_spec(self, **kw):
        return "UUID"

    def bind_processor(self, dialect):
        def process(value):
            if value is not None:
                value = str(value)
            return value

        return process

    def result_processor(self, dialect, coltype):
        def process(value):
            if value is not None:
                value = uuid.UUID(value)
            return value

        return process
```
